src/Twig/Extension/TemplateRegistryExtension.php

```diff
@@ -11,59 +11,82 @@
 
 namespace Sonata\AdminBundle\Twig\Extension;
 
-use Sonata\AdminBundle\Admin\AdminInterface;
-use Sonata\AdminBundle\Admin\Pool;
+use Sonata\AdminBundle\Templating\TemplateRegistryInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Exception\ServiceCircularReferenceException;
+use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
 use Twig\Extension\AbstractExtension;
 use Twig\TwigFunction;
 
 final class TemplateRegistryExtension extends AbstractExtension
 {
     /**
-     * @var Pool
+     * @var TemplateRegistryInterface
      */
-    private $pool;
+    private $globalTemplateRegistry;
 
-    public function __construct(Pool $pool)
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    public function __construct(TemplateRegistryInterface $globalTemplateRegistry, ContainerInterface $container)
     {
-        $this->pool = $pool;
+        $this->globalTemplateRegistry = $globalTemplateRegistry;
+        $this->container = $container;
     }
 
     public function getFunctions()
     {
         return [
             new TwigFunction('get_admin_template', [$this, 'getAdminTemplate']),
-            new TwigFunction('get_admin_pool_template', [$this, 'getPoolTemplate']),
+            new TwigFunction('get_global_template', [$this, 'getGlobalTemplate']),
+
+            // NEXT MAJOR: Remove this line
+            new TwigFunction('get_admin_pool_template', [$this, 'getGlobalTemplate'], ['deprecated' => true]),
         ];
     }
 
     /**
      * @param string $name
      * @param string $adminCode
      *
+     * @throws ServiceNotFoundException
+     * @throws ServiceCircularReferenceException
+     *
      * @return null|string
      */
     public function getAdminTemplate($name, $adminCode)
     {
-        return $this->getAdmin($adminCode)->getTemplate($name);
+        return $this->getTemplateRegistry($adminCode)->getTemplate($name);
     }
 
     /**
      * @param string $name
      *
      * @return null|string
      */
-    public function getPoolTemplate($name)
```

Comment:
isnt this a BC break?

Reply:
You're right. I'll put the original function back and deprecate it.